homeassistant/components/template/config.py

```diff
@@ -39,6 +58,7 @@
         vol.Optional(CONF_TRIGGER): cv.TRIGGER_SCHEMA,
```

Comment:
As it stands, you can't use Jinja templates in triggers. It would be great to be able to do so, but i think that would mean moving the logic for registering triggers much later. Is that OK?

Here's an example:
```yaml
blueprint:
  domain: template
  input:
    tracked_entity:
      name: Tracked entity
      description: The `sensor` which needs to change from `something` to `something_else` for the resulting `binary_sensor` to turn `on`
      selector:
        entity:
variables:
  tracked_entity: !input tracked_entity
trigger:
  - platform: state
    entity_id: "{{ tracked_entity }}" # error: `Invalid config for 'template' at configuration.yaml, line 88: Entity {{ tracked_entity }} is neither a valid entity ID nor a valid UUID for dictionary value 'entity_id', got None`
    from: "something"
    to: "something_else"
binary_sensor:
  ...
```

`trigger_variables` might be needed, but the question is when should they be evaluated? Is it safe to do so at config validation? Or should the trigger registration be postponed to much later, when it's safe to evaluate `trigger_variables` as regular tempaltes?

Reply:
This isn't specific to blueprints, right? If it's not specific to blueprints I suggest discussing it separately in a new architecture discussion.

Reply:
I'd say that template blueprints expose the missing feature. In automations, you can use `trigger_variables` specifically for allowing the templating of `entity_id` in blueprints, which had not existed for templates until now.

> The second variant is setting variables that are available when attaching a trigger when the trigger can contain templated values. These are defined using the trigger_variables key at an automation level. These variables can only contain [limited templates](https://www.home-assistant.io/docs/configuration/templating/#limited-templates). The triggers will not re-apply if the value of the template changes. Trigger variables are a feature meant to support using blueprint inputs in triggers.

([source](https://www.home-assistant.io/docs/automation/trigger/#trigger-variables))

I'm a but stumped about what to do here. If you think this discussion is outside the scope of this feature, I'll start a new architecture discussion, but again... template blueprints _uncovered_ this issue, so i'd say it's related.

Reply:
I don't think it'd related since it's about triggers and not blueprints.

Reply:
We don't need to solve all problems at once.

Reply:
https://github.com/home-assistant/architecture/discussions/1153